### PR TITLE
Choice replace false

### PR DESCRIPTION
**Pull request by unnonouno**
I made a naive implementation of choice with `replace=False` and without `p`.

### CODE REVIEW

File: tests/cupy_tests/random_tests/test_generator.py [hunk 1]
```diff
@@ -399,6 +399,60 @@ def test_choice_invalid_value(self):
...
+    def test_dtype_shape(self):
+        v = self.rs.choice(a=self.a, size=self.size, replace=False)
+        if isinstance(self.size, six.integer_types):
+            expected_shape = (self.size,)
+        else:
+            expected_shape = self.size
+        if isinstance(self.a, numpy.ndarray):
+            expected_dtype = 'float'
+        else:
+            expected_dtype = 'int64'
```
Comment (kmatsuura) [inline tests/cupy_tests/random_tests/test_generator.py]: This test has no error using 'int' instead of 'int64' on both of ubuntu and Windows.

Comment (unnonouno) [reply tests/cupy_tests/random_tests/test_generator.py]: ok

File: cupy/random/generator.py [hunk 2]
```diff
@@ -272,14 +272,26 @@ def choice(self, a, size=None, replace=True, p=None):
...
         if size is None:
             raise NotImplementedError
         shape = size
         size = numpy.prod(shape)
 
+        if not replace and p is None:
+            if a_size < size:
+                raise ValueError(
+                    'Cannot take a larger sample than population when '
+                    '\'replace=False\'')
```
Comment (okuta) [inline cupy/random/generator.py]: How about using `"`?

**Comment by okuta** [comment]
LGTM except comment.

File: cupy/random/generator.py (hunk 2 above)
Comment (unnonouno) [reply cupy/random/generator.py]: I used single quotes for consistency. Is that fix important?

Comment (okuta) [reply cupy/random/generator.py]: OK. Not important.

**Pull request merged**